src/System.Net.Security/src/System/Net/_SslSessionsCache.cs

```diff
@@ -53,7 +53,7 @@ internal SslCredKey(byte[] thumbPrint, int allowedProtocols, bool serverMode, En
 
                 _HashCode ^= allowedProtocols;
                 _HashCode ^= (int)encryptionPolicy;
-                _HashCode ^= serverMode ? 5 : 7; //TODO (Issue #3362) used a prime number here as it's a XOR. Figure out appropriate value.
+                _HashCode ^= serverMode ? 5 : 7; // Used a prime number here as it's a XOR. The values should not conflict with EncryptionPolicy and other parts of the hash
```

Comment:
I'm not really certain why it's setting the 0 and 2 bits, and sometimes also the 1 bit.  `serverMode ? 0 : 2` is maybe better, and trying to twiddle some bits that aren't twiddled by encryptionPolicy or allowedProtocols is best (unless your hash table uses a power-of-two of buckets, in which case only the low bits matter).